src/OrchardCore/OrchardCore/Shell/ShellHost.cs

```diff
@@ -467,6 +496,25 @@ private bool CanReleaseShell(ShellSettings settings)
             return settings.State != TenantState.Disabled || _shellContexts.TryGetValue(settings.Name, out var value) && value.ActiveScopes == 0;
         }
 
+        /// <summary>
+        /// Checks if a shell can be removed, throws an exception if the shell is neither uninitialized nor disabled.
+        /// </summary>
+        private void CheckCanRemoveShell(ShellSettings settings)
+        {
+            if (settings.Name == ShellHelper.DefaultShellName)
+            {
+                throw new InvalidOperationException($"The '{ShellHelper.DefaultShellName}' tenant can't be removed.");
+            }
+
+            if (settings.State != TenantState.Uninitialized &&
```

Comment:
I noticed the reuse of the same condition many places. Maybe we should create an extension for this?

```
    public static bool IsRemovableState(this ShellSettings shellSettings) =>
        shellSettings.State == TenantState.Uninitialized ||
        shellSettings.State == TenantState.Disabled;
```

Reply:
Yes I agree, there are many others that could be created, from memory in another recent PR I created an `IsNotIninialized()` extension checking that the tenant is not Running and not Disabled.

So if worth to do maybe in a separate and dedicated PR for all these extensions and use them, or let people create their own extensions (or not) for their own usage.

But no against at all, if worth in a dedicated PR in one pass.

Reply:
Well, I think meanwhile you can extract the logic used by this PR into an extension. It's up to you.

Reply:
Okay, I will check it.

But not here in this private `CheckCanRemoveShell()`, because here the AND is not only used on `Disabled` but on `(Disabled AND ActiveScopes > 0)`.

            if (settings.State != TenantState.Uninitialized &&
                (settings.State != TenantState.Disabled ||
                _shellContexts.TryGetValue(settings.Name, out var value) && value.ActiveScopes > 0))

Reply:
Done, I named it IsRemovable()